src/main/scala/io/druid/indexer/spark/SparkDruidIndexer.scala

```diff
@@ -636,24 +641,19 @@ class DateBucketAndHashPartitioner(gran: Granularity, partMap: Map[(Long, Long),
           throw new IAE(s"Unknown value type ${x.getClass} : [$x]")
       }
       val dateBucket = gran.truncate(new DateTime(k)).getMillis
-      val modSize = maxTimePerBucket.get(dateBucket.toLong)
-      if (modSize.isEmpty) {
-        // Lazy ISE creation
-        throw new ISE(s"bad date bucket [${dateBucket.toLong}]. " +
-          s"available: ${maxTimePerBucket.keySet}")
+      val shardLookup = shardLookups.get(dateBucket)
+      if (shardLookup.isEmpty) {
+        throw new IAE(s"Bad date bucket $dateBucket")
       }
-      val m = modSize.get
-      if (m <= 0) {
-        throw new ISE(s"Illegal mod [$m]")
+      val shardSpec = Option(shardLookup.get.getShardSpec(dateBucket, new MapBasedInputRow(dateBucket, if (dims == null) map.keys.toList.asJava else dims.toList.asJava, map.asJava)))
```

Comment:
there's little point in using an option here, you might as well check null instead of isEmpty, it will make the later code more readable. Alternatively you can do `val shardSpec = Option(...) orElse { throw new ISE(...) }` to make it flow better, but the option has little value beyond this point.